### PR TITLE
multiplayer enigma boogaloo

### PR DESCRIPTION
**Pull request by jaskarth**
Worldgen mappings, as well as a few misc mappings, by SuperCoder79, Earthcomputer, and Shedaniel.

### CODE REVIEW

File: mappings/net/minecraft/client/font/TextRenderer.mapping [hunk 1]
```diff
@@ -30,9 +31,15 @@ CLASS net/minecraft/class_327 net/minecraft/client/font/TextRenderer
 	METHOD method_1719 setRightToLeft (Z)V
 		ARG 1 rightToLeft
 	METHOD method_1720 drawWithShadow (Lnet/minecraft/class_4587;Ljava/lang/String;FFI)I
+		ARG 1 matrixes
```
Comment (YanisBft) [inline mappings/net/minecraft/client/font/TextRenderer.mapping]: `matrices`

File: mappings/net/minecraft/client/font/TextRenderer.mapping [hunk 2]
```diff
@@ -55,16 +62,24 @@ CLASS net/minecraft/class_327 net/minecraft/client/font/TextRenderer
 		ARG 10 light
 	METHOD method_1726 isRightToLeft ()Z
 	METHOD method_1727 getStringWidth (Ljava/lang/String;)I
+		ARG 1 text
 	METHOD method_1728 wrapStringToWidthAsList (Lnet/minecraft/class_2561;I)Ljava/util/List;
 		ARG 2 width
 	METHOD method_1729 draw (Lnet/minecraft/class_4587;Ljava/lang/String;FFI)I
+		ARG 1 matrixes
```
Comment (YanisBft) [inline mappings/net/minecraft/client/font/TextRenderer.mapping]: `matrices`

File: mappings/net/minecraft/client/font/TextRenderer.mapping [hunk 3]
```diff
@@ -74,3 +89,43 @@ CLASS net/minecraft/class_327 net/minecraft/client/font/TextRenderer
 		ARG 8 seeThrough
 		ARG 9 backgroundColor
 		ARG 10 light
+	METHOD method_27517 drawWithShadow (Lnet/minecraft/class_4587;Lnet/minecraft/class_2561;FFI)I
+		ARG 1 matrixes
```
Comment (YanisBft) [inline mappings/net/minecraft/client/font/TextRenderer.mapping]: `matrices`

File: mappings/net/minecraft/client/font/TextRenderer.mapping [hunk 4]
```diff
@@ -74,3 +89,43 @@ CLASS net/minecraft/class_327 net/minecraft/client/font/TextRenderer
...
+		ARG 2 x
+		ARG 3 y
+		ARG 4 color
+		ARG 5 matrix
+		ARG 6 shadow
+		ARG 7 mirror
+	METHOD method_27525 getStringWidth (Lnet/minecraft/class_2561;)I
+		ARG 1 text
+	METHOD method_27528 draw (Lnet/minecraft/class_4587;Lnet/minecraft/class_2561;FFI)I
+		ARG 1 matrixes
```
Comment (YanisBft) [inline mappings/net/minecraft/client/font/TextRenderer.mapping]: `matrices`

File: mappings/net/minecraft/world/gen/trunk/LargeOakTrunkPlacer.mapping [hunk 5]
```diff
@@ -0,0 +1,28 @@
+CLASS net/minecraft/class_5212 net/minecraft/world/gen/trunk/LargeOakTrunkPlacer
+	METHOD method_27391 isHighEnough (II)Z
+		ARG 1 treeHeight
+		ARG 2 height
+	METHOD method_27392 makeBranches (Lnet/minecraft/class_3747;Ljava/util/Random;ILnet/minecraft/class_2338;Ljava/util/List;Ljava/util/Set;Lnet/minecraft/class_3341;Lnet/minecraft/class_4643;)V
+		ARG 1 world
+		ARG 3 treeHeight
+		ARG 4 treePosition
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/trunk/LargeOakTrunkPlacer.mapping]: `treePos`

File: mappings/net/minecraft/util/math/noise/DoublePerlinNoiseSampler.mapping [hunk 6]
```diff
@@ -0,0 +1,11 @@
+CLASS net/minecraft/class_5216 net/minecraft/util/math/noise/DoublePerlinNoiseSampler
+	FIELD field_24175 amplitude D
+	FIELD field_24176 sampler1 Lnet/minecraft/class_3537;
+	FIELD field_24177 sampler2 Lnet/minecraft/class_3537;
```
Comment (liach) [inline mappings/net/minecraft/util/math/noise/DoublePerlinNoiseSampler.mapping]: first/second sampler

File: mappings/net/minecraft/net/minecraft/world/gen/feature/size/ThreeLayersFeatureSize.mapping [hunk 7]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/class_5203 net/minecraft/net/minecraft/world/gen/feature/size/ThreeLayersFeatureSize
```
Comment (YanisBft) [inline mappings/net/minecraft/net/minecraft/world/gen/feature/size/ThreeLayersFeatureSize.mapping]: Didn't spot that at first look, but wtf

Comment (jaskarth) [reply mappings/net/minecraft/net/minecraft/world/gen/feature/size/ThreeLayersFeatureSize.mapping]: That is the registry name

Comment (jaskarth) [reply mappings/net/minecraft/net/minecraft/world/gen/feature/size/ThreeLayersFeatureSize.mapping]: oh oops i did a fucky wucky hold on

File: mappings/net/minecraft/world/gen/trunk/LargeOakTrunkPlacer.mapping [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+		ARG 4 treePos
+		ARG 5 branches
+		ARG 8 config
+	METHOD method_27393 makeOrCheckBranch (Lnet/minecraft/class_3747;Ljava/util/Random;Lnet/minecraft/class_2338;Lnet/minecraft/class_2338;ZLjava/util/Set;Lnet/minecraft/class_3341;Lnet/minecraft/class_4643;)Z
+		ARG 1 world
+		ARG 3 start
+		ARG 4 end
+		ARG 5 make
+		ARG 8 config
+	METHOD method_27394 getLongsetSide (Lnet/minecraft/class_2338;)I
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/trunk/LargeOakTrunkPlacer.mapping]: Is "longset" a word?

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/client/font/TextRenderer.mapping [hunk 9]
```diff
@@ -163,7 +163,35 @@ CLASS net/minecraft/class_327 net/minecraft/client/font/TextRenderer
 		ARG 8 seeThrough
 		ARG 9 backgroundColor
 		ARG 10 light
-		ARG 11 rightToLeft
+	METHOD method_27517 drawWithShadow (Lnet/minecraft/class_4587;Lnet/minecraft/class_2561;FFI)I
```
Comment (liach) [inline mappings/net/minecraft/client/font/TextRenderer.mapping:167]:
```suggestion
		ARG 11 rightToLeft
	METHOD method_27517 drawWithShadow (Lnet/minecraft/class_4587;Lnet/minecraft/class_2561;FFI)I
```

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/font/TextRenderer.mapping (hunk 9 above)
Comment (YanisBft) [reply mappings/net/minecraft/client/font/TextRenderer.mapping:167]: ??

Comment (liach) [reply mappings/net/minecraft/client/font/TextRenderer.mapping:167]: Restore the ARG 11 or the previous method

**Review by Runemoro — APPROVED** [review]

**Pull request merged**